AudioKit/Common/Internals/AudioKit.swift

```diff
@@ -205,79 +212,82 @@ public typealias AKCallback = () -> Void
+                })
+                notificationObservers.append(routeChangeObserver)
 
-            try engine.start()
-            shouldBeRunning = true
+            }
 
-        } catch {
-            fatalError("AudioKit: Could not start engine. error: \(error).")
-        }
+            if AKSettings.enableCategoryChangeHandling {
+                let configurationChangeObserver = NotificationCenter.default.addObserver(forName: .AVAudioEngineConfigurationChange,
+                                                                                         object: engine,
+                                                                                         queue: OperationQueue.main,
+                                                                                         using: { (notification) in
+                                                                                            AudioKit.restartEngineAfterConfigurationChange(notification)
+                })
+                notificationObservers.append(configurationChangeObserver)
+            }
+            try updateSessionCategoryAndOptions()
+            try AVAudioSession.sharedInstance().setActive(true)
+        #endif
+
+        try engine.start()
+        shouldBeRunning = true
     }
 
-    @objc fileprivate static func updateSessionCategoryAndOptions() {
+    @objc fileprivate static func updateSessionCategoryAndOptions() throws {
         #if !os(macOS)
-            do {
-                let sessionCategory = AKSettings.computedSessionCategory()
-                let sessionOptions = AKSettings.computedSessionOptions()
-
-                #if os(iOS)
-                    try AKSettings.setSession(category: sessionCategory,
-                                              with: sessionOptions)
-                #elseif os(tvOS)
-                    try AKSettings.setSession(category: sessionCategory)
-                #endif
-            } catch {
-                fatalError("AudioKit: Could not update AVAudioSession category and options. error: \(error).")
-            }
+            let sessionCategory = AKSettings.computedSessionCategory()
+            let sessionOptions = AKSettings.computedSessionOptions()
+
+            #if os(iOS)
+                try AKSettings.setSession(category: sessionCategory,
+                                          with: sessionOptions)
+            #elseif os(tvOS)
+                try AKSettings.setSession(category: sessionCategory)
+            #endif
         #endif
     }
 
     /// Stop the audio engine
-    @objc open static func stop() {
+    @objc open static func stop() throws {
         // Stop the engine.
         engine.stop()
         shouldBeRunning = false
+
         #if os(iOS)
+        notificationObservers.forEach { (observer) in
+            NotificationCenter.default.removeObserver(observer)
+        }
+
         do {
```

Comment:
You could remove this do-catch, just let the error message bubble up to the user if not able to recover internally.